echoregions/regions2d/regions2d.py

```diff
@@ -358,122 +356,129 @@ def plot(
 
-        # Convert region time to integer timestamp.
-        df["time"] = matplotlib.dates.date2num(df["time"])
+        if region_df.empty:
+            print(
+                "Post NaN Depth Filtered Regions is Empty. All rows had NaN Depth values."
+            )
+        else:
+            # Organize the regions in a format for region mask.
+            df = region_df.explode(["time", "depth"])
 
-        # Create a list of dataframes for each regions.
-        grouped = list(df.groupby("region_id"))
+            # Convert region time to integer timestamp.
+            df["time"] = matplotlib.dates.date2num(df["time"])
 
-        # Convert to list of numpy arrays which is an acceptable format to create region mask.
-        regions_np = [np.array(region[["time", "depth"]]) for id, region in grouped]
+            # Create a list of dataframes for each regions.
+            grouped = list(df.groupby("region_id"))
 
-        # Corresponding region ids converted to int.
-        region_ids = [int(id) for id, region in grouped]
+            # Convert to list of numpy arrays which is an acceptable format to create region mask.
+            regions_np = [np.array(region[["time", "depth"]]) for _, region in grouped]
 
-        # Convert ping_time to unix_time since the masking does not work on datetime objects.
-        da_Sv = da_Sv.assign_coords(
-            unix_time=(
-                "ping_time",
-                matplotlib.dates.date2num(da_Sv.coords["ping_time"].values),
-            )
-        )
+            # Corresponding region ids converted to int.
+            filtered_region_id = [int(id) for id, _ in grouped]
 
-        # Set up mask labels.
-        if mask_labels == "from_ids":
-            r = regionmask.Regions(outlines=regions_np, numbers=region_ids)
-        elif isinstance(mask_labels, list) or mask_labels is None:
-            r = regionmask.Regions(outlines=regions_np)
-        else:
-            raise ValueError("mask_labels must be None, 'from_ids', or a list.")
+            # Convert ping_time to unix_time since the masking does not work on datetime objects.
+            da_Sv = da_Sv.assign_coords(
+                unix_time=(
+                    "ping_time",
+                    matplotlib.dates.date2num(da_Sv.coords["ping_time"].values),
+                )
+            )
 
-        # Create mask
-        try:
-            M = r.mask(
+            # Create mask
+            r = regionmask.Regions(
+                outlines=regions_np, names=filtered_region_id, name=mask_name
+            )
+            mask_3d = r.mask_3D(
                 da_Sv["unix_time"],
                 da_Sv["depth"],
                 wrap_lon=False,
+            ).astype(
+                int
+            )  # This maps False to 0 and True to 1
+
+            # Replace region coords with region_id coords
+            mask_3d = mask_3d.assign_coords(region_id=mask_3d["names"])
+            mask_3d = mask_3d.swap_dims({"region": "region_id"})
```

Comment:
These 2 lines effectively is `mask_3d = mask_3d.rename({"names": "region_id"})`, is this correct? since `names` is already aligned with `region`. If so, use that to avoid confusion.

Reply:
Not exactly, since I still would need to swap the actual dimension, which is set in regionmask as `region` not `names`.

I'm using the following instead since `assign_coords` is somewhat redundant since `names` is what we need here for region IDs:
```
mask_3d = mask_3d.rename({"names": "region_id"})
mask_3d = mask_3d.swap_dims({"region": "region_id"})
```